packages/platform-android/native_modules.gradle

```diff
@@ -0,0 +1,250 @@
+   * a `ext.reactNativeProjectRoot` extension property
+   *
+   * @return
+   */
+  File getReactNativeProjectRoot() {
+    if (this.extension.has("reactNativeProjectRoot")) {
+      File rnRoot = File(this.extension.get("reactNativeProjectRoot"))
+      // allow custom React Native project roots for non-standard directory structures
+      this.logger.debug("${LOG_PREFIX}Using custom React Native project root path '${rnRoot.toString()}'")
+      return rnRoot
+    }
+
+    File androidRoot
+
+    if (this.project) {
+      androidRoot = this.project.rootProject.projectDir
+    } else {
+      androidRoot = this.defaultSettings.rootProject.projectDir
+    }
+
+    this.logger.debug("${LOG_PREFIX}Using default React Native project root path '${androidRoot.parentFile.toString()}'")
+    return androidRoot.parentFile
+  }
+
+  /**
+   * Code-gen a java file with all the detected ReactNativePackage instances automatically added
+   *
+   * @param outputDir
+   * @param generatedFileName
+   * @param generatedFileContentsTemplate
+   * @param applicationId
+   */
+  void generatePackagesFile(File outputDir, String generatedFileName, String generatedFileContentsTemplate, String applicationId) {
+    ArrayList<HashMap<String, String>>[] packages = this.reactNativeModules
+
+    String packageImports = ""
+    String packageClassInstances = ""
+
+    if (packages.size() > 0) {
+      packageImports = "import ${applicationId}.BuildConfig;\n\n"
+      packageImports = packageImports + packages.collect {
+        "// ${it.name}\n${it.packageImportPath}"
+      }.join(';\n')
+      packageClassInstances = ",\n      " + packages.collect { it.packageInstance }.join(',')
+    }
+
+    String generatedFileContents = generatedFileContentsTemplate
+      .replace("{{ packageImports }}", packageImports)
+      .replace("{{ packageClassInstances }}", packageClassInstances)
+
+    outputDir.mkdirs()
+    final FileTreeBuilder treeBuilder = new FileTreeBuilder(outputDir)
+    treeBuilder.file(generatedFileName).newWriter().withWriter { w ->
+      w << generatedFileContents
+    }
+  }
+
+  /**
+   * Runs a process to call the React Native CLI Config command and parses the output
+   *
+   * @return ArrayList < HashMap < String , String > >
+   */
+  ArrayList<HashMap<String, String>> getReactNativeConfig() {
+    if (this.reactNativeModules != null) return this.reactNativeModules
+    ArrayList<HashMap<String, String>> reactNativeModules = new ArrayList<HashMap<String, String>>()
+
+    def cmdProcess
+
+    try {
+      cmdProcess = Runtime.getRuntime().exec(REACT_NATIVE_CONFIG_CMD, null, getReactNativeProjectRoot())
+      cmdProcess.waitFor()
+    } catch (Exception exception) {
+      this.logger.warn("${LOG_PREFIX}${exception.message}")
+      this.logger.warn("${LOG_PREFIX}Automatic import of native modules failed. (UNKNOWN)")
+      return reactNativeModules
+    }
+
+    def reactNativeConfigOutput = cmdProcess.in.text
```

Comment:
does this take output from `stdout` only or `stderr` is there as well? `config` outputs warnings in `stderr` so we need to either only take stdout or filter out the warnings like iOS implementation does (because `execute_command` cannot into stdout only)

Reply:
`stdout` only - `.in` is short hand for `getInputStream()`, `getErrorStream()` is where `stderr` is piped to

Reply:
That's what I assumed, but wanted to be sure. Thanks! Too bad we don't have cross-platform API for that in Ruby and CocoaPods :< (or at least I don't know about them and quick google didn't reveal there's such; cc @orta)

Reply:
There’s definitely APIs for that and `execute_command` uses it under the hood, but it doesn’t currently expose a way to keep that output separate, which it probably _should_, but that would be an iteration.